src/pages/timecard/Timecard.spec.jsx

```diff
@@ -82,9 +82,32 @@ describe('Timecard', () => {
 
     expect(screen.queryByText('Add a new time period')).toBeFalsy();
     expect(screen.queryByText('Shift')).toBeFalsy();
+    expect(screen.queryByText('Non-working day')).toBeFalsy();
     expect(screen.getByText('Scheduled rest day')).toBeTruthy();
   });
 
+  it('should render the NonWorkingDay component when time period type is NWD', async () => {
+    renderWithTimecardContext(<Timecard />, {
+      summaryErrors: {},
+      setSummaryErrors: jest.fn(),
+      timeEntries: [
+        {
+          timePeriodTypeId: '00000000-0000-0000-0000-000000000003',
+          startTime: '',
+          finishTime: '',
+        },
+      ],
+      setTimeEntries: jest.fn(),
+      timecardDate: '',
+      setTimecardDate: jest.fn(),
+    });
+
+    expect(screen.queryByText('Add a new time period')).toBeFalsy();
+    expect(screen.queryByText('Shift')).toBeFalsy();
+    expect(screen.queryByText('Scheduled rest day')).toBeFalsy();
+    expect(screen.getByText('Non-working day')).toBeTruthy();
```

Comment:
Since we're implementing more time period types now we'll have to keep remembering to add each type to these tests. 
It might be better to loop though the time periods and test them all which will keep the tests up to date when we implement more periods.

Could do something like this if you import the time periods from the mock data and then apply to the shift, SRD and this test:
```suggestion
  const testSimpleTimePeriodRendersWithCorrectTitle = (timePeriodTypeTitle) => {
    expect(screen.getByText(timePeriodTypeTitle)).toBeTruthy();
    timeCardPeriodTypes.forEach(type => {
      if (type.name != timePeriodTypeTitle) expect(screen.queryByText(type.name)).toBeFalsy();
  })
}
```

Reply:
Done